packages/svelte/src/compiler/phases/2-analyze/css/css-analyze.js

```diff
@@ -184,6 +221,11 @@ const css_visitors = {
 
 				if (idx !== -1) {
 					is_global_block = true;
+					if (i === 0) {
+						if (is_unscoped_global(context.path) && node.block.children.length > 0) {
+							context.state.has_global_unscoped.value = true;
```

Comment:
once this has flipped to true you don't need to evaluate more rules

Reply:
correct...i will still need to loop to do the rest of the analysis but i can guard the usage if it's already true...fixed it in the last commit